File: Back/src/main/java/com/example/back/Services/EvaluationService.java
package com.example.back.Services;

import com.example.back.Entities.Evaluation;
import com.example.back.Entities.Tache_journal;

import java.util.List;
import java.util.Set;

public interface EvaluationService {

    Set<Evaluation> addEvaluationAndAssignToJournal (long id_Journal);

    List<Evaluation> findAllEvaluationsByIdJournal(Long id_Journal);


    Evaluation findEvaluationById(Long idEvaluation);

    List<Evaluation> findAll();

    Long updateEvaluation(Long idEvaluation , Evaluation updtevaluation ) ;


}

File: Front/src/app/Services/EvaluationService/evaluation.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";

import {Evaluation}   from "../../Modules/EvaluationModule/Evaluation.module";
import {Observable} from "rxjs";
import {TacheJournal} from "../../Modules/TacheJournalModule/TacheJournal.module";
import {Journal} from "../../Modules/JournalModule/Journal.module";

@Injectable({
  providedIn: 'root'
})
export class EvaluationService {

  private apiUrl = 'http://localhost:9090/api/services/evaluation';

  constructor(private http: HttpClient)  { }



  findAllEvaluationsByIdJournal(id: number): Observable<Evaluation[]> {
    return this.http.get<Evaluation[]>(`http://localhost:9090/api/services/evaluation/GetEvaluationsbyJournal/${id}`);
  }

  updateEvaluation(idEvaluation: number, evaluation: Evaluation): Observable<Journal> {
    const url = `${this.apiUrl}/updatEvaluation/${idEvaluation}`;
    return this.http.put<Journal>(url, evaluation);
  }
}
